WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift

```diff
@@ -0,0 +1,109 @@
+import UIKit
+import Yosemite
+import Combine
+
+/// View model for `BlazeCampaignDashboardView`.
+final class BlazeCampaignDashboardViewModel: ObservableObject {
+    /// UI state of the Blaze campaign view in dashboard.
+    enum State {
+        /// Shows placeholder views in redacted state.
+        case loading
+        /// Shows info about the latest Blaze campaign
+        case showCampaign(campaign: BlazeCampaign)
+        /// Shows info about the latest published Product
+        case showProduct(product: Product)
+        /// When there is no campaign or published product
+        case empty
+    }
+
+    @Published private(set) var state: State
+
+    @Published private(set) var shouldShowInDashboard: Bool = false
+
+    private(set) var isRedacted: Bool = true
+
+    var shouldShowShowAllCampaignsButton: Bool {
+        if case .showCampaign(_) = state {
```

Comment:
Nit: you can omit the closure if you don't care about the associated value.

```suggestion
        if case .showCampaign = state {
```

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/10958/commits/f96ef481e1ae2a312bac87a5fdb1e14d6448c922